tests/foreman/virtwho/test_ui_virtwho.py

```diff
@@ -420,3 +428,201 @@ def test_positive_virtwho_configs_widget(session, form_data):
+        deploy_configure_by_command(config_command)
+        assert session.virtwho_configure.search(name)[0]['Status'] == 'ok'
+        session.virtwho_configure.delete(name)
+        assert not session.virtwho_configure.search(name)
+        restart_virt_who_service()
+        assert get_virtwho_status() == 'logerror'
+
+
+@tier2
+def test_positive_virt_who_reporter_permission(session, test_name, form_data):
+    """Verify the virt-who reporter role can TRULY work.
+
+    :id: 4bef3b31-89d2-4502-b971-2437b920a3ff
+
+    :expectedresults:
+        Virt-who Reporter Role granting minimal set of permissions for virt-who
+        to upload the report, it can be used if you configure virt-who manually
+        and want to use user that has locked down account.
+    """
+    username = gen_string('alpha')
+    password = gen_string('alpha')
+    with session:
+        # Create an user
+        session.user.create({
+            'user.login': username,
+            'user.mail': 'test@test.com',
+            'user.auth': 'INTERNAL',
+            'user.password': password,
+            'user.confirm': password,
+        })
+        # Create a virt-who config plugin
+        virt_who_name = gen_string('alpha')
+        form_data['name'] = virt_who_name
+        session.virtwho_configure.create(form_data)
+        values = session.virtwho_configure.read(virt_who_name)
+        command = values['deploy']['command']
+        deploy_configure_by_command(command, debug=True)
+        assert session.virtwho_configure.search(virt_who_name)[0]['Status'] == 'ok'
+        # Update the virt-who config file
+        config_id = get_configure_id(virt_who_name)
+        update_configure_option('rhsm_username', username, config_id)
+        delete_configure_option('rhsm_encrypted_password', config_id)
+        add_configure_option('rhsm_password', password, config_id)
+        restart_virt_who_service()
+        assert get_virtwho_status() == 'logerror'
+        # Check the permissioin of Virt-who Reporter
+        session.user.update(
+            username,
+            {'roles.resources.assigned': ['Virt-who Reporter']}
+        )
+        assert session.user.search(username)[0]['Username'] == username
+        user = session.user.read(username)
+        assert user['roles']['resources']['assigned'] == ['Virt-who Reporter']
+        restart_virt_who_service()
+        assert get_virtwho_status() == 'running'
+        with Session(test_name, username, password) as newsession:
+            try:
+                newsession.virtwho_configure.read(virt_who_name)
+            except Exception:
+                pass
+            else:
+                raise VirtWhoError(
+                    'Virt-who Reporter should not have the permission to read config')
+        session.user.delete(username)
+        assert not session.user.search(username)
+
+
+def test_positive_virt_who_viewer_permission(session, test_name, form_data):
```

Comment:
rename the case: test_positive_virtwho_viewer_role()

Reply:
ok